dvc/fs/webhdfs.py

```diff
@@ -66,115 +32,71 @@ def _get_kwargs_from_urls(urlpath):
+                return None
+
+        if config.has_section(alias):
+            return dict(config.items(alias))
+
+    def _prepare_credentials(self, **config):
+        hdfs_config = self._parse_config(
+            path=config.pop("hdfscli_config", None),
+            alias=config.pop("webhdfs_alias", None),
+        )
+        if hdfs_config is not None:
+            # If this is not None, then we will solely use
+            # config options from this.
+
+            if "url" in hdfs_config:
+                url = urlparse(hdfs_config.pop("url"))
+                hdfs_config["use_ssl"] = url.scheme == "https"
+                webhdfs_url = urlunparse(url._replace(scheme="webhdfs"))
+
+                # override URL options (such as user) with the explicitly
+                # given ones
+                hdfs_config = {
+                    **self._get_kwargs_from_urls(webhdfs_url),
+                    **hdfs_config,
+                }
+
+            if "root" in hdfs_config:
+                raise ValueError(
+                    '"root" option is not supported any '
+                    "more for webhdfs remotes."
                 )
 
-        return client
-
-    @contextmanager
-    def open(self, path_info, mode="r", encoding=None, **kwargs):
-        assert mode in {"r", "rt", "rb"}
-
-        with self.hdfs_client.read(
-            path_info.path, encoding=encoding
-        ) as reader:
-            yield reader
-
-    def walk_files(self, path_info, **kwargs):
-        if not self.exists(path_info):
-            return
+            return hdfs_config
 
-        root = path_info.path
-        for path, _, fnames in self.hdfs_client.walk(root):
-            for fname in fnames:
-                yield path_info.replace(path=posixpath.join(path, fname))
+        if "webhdfs_token" in config:
+            config["token"] = config.pop("webhdfs_token")
 
-    def remove(self, path_info):
-        if path_info.scheme != self.scheme:
-            raise NotImplementedError
+        return config
 
-        self.hdfs_client.delete(path_info.path)
-
-    def exists(self, path_info) -> bool:
-        assert not isinstance(path_info, list)
-        assert path_info.scheme == "webhdfs"
-
-        status = self.hdfs_client.status(path_info.path, strict=False)
-        return status is not None
+    @wrap_prop(threading.Lock())
+    @cached_property
+    def fs(self):
+        from fsspec.implementations.webhdfs import WebHDFS
 
-    def info(self, path_info):
-        st = self.hdfs_client.status(path_info.path)
-        return {"size": st["length"], "type": "file"}
+        return WebHDFS(**self.fs_args)
 
     def checksum(self, path_info):
-        return HashInfo(
-            "checksum",
-            self.hdfs_client.checksum(path_info.path)["bytes"],
-            size=self.hdfs_client.status(path_info.path)["length"],
-        )
+        path = self._with_bucket(path_info)
+        ukey = self.fs.ukey(path)
```

Comment:
Will need to properly distinguish between `checksum` and `ukey` in dvc in the future. I did a mistake of using `checksum` for refodb in staging, while we could really use ukey there instead.